mempool/src/shared_mempool/network.rs

```diff
@@ -578,11 +586,18 @@ fn compare_prioritized_peers(
             let role_a = peer_a.1;
             let role_b = peer_b.1;
             match role_a.cmp(&role_b) {
-                // Then tiebreak by PeerId for stability
+                // Tiebreak by hash(PeerId). Stable within a mempool instance but random between instances.
                 Ordering::Equal => {
-                    let peer_id_a = peer_network_id_a.peer_id();
-                    let peer_id_b = peer_network_id_b.peer_id();
-                    peer_id_a.cmp(&peer_id_b)
+                    let mut hasher_a = random_state.build_hasher();
+                    let mut hasher_b = random_state.build_hasher();
```

Comment:
it's probably better to have a helper function that calculate the hash for some T than this